mcux/CMakeLists.txt

```diff
@@ -72,12 +76,23 @@ zephyr_compile_definitions(${MCUX_CPU})
 # Build mcux device-specific objects. Although it is not normal
 # practice, drilling down like this avoids the need for repetitive
 # build scripts for every mcux device.
+
 zephyr_library_sources(mcux-sdk/devices/${MCUX_DEVICE_PATH}/drivers/fsl_clock.c)
 if (${MCUX_DEVICE} MATCHES "LPC|MIMXRT6|MIMXRT5")
-  zephyr_library_sources(mcux-sdk/devices/${MCUX_DEVICE_PATH}/drivers/fsl_power.c)
+  # fsl_power.c does not build on DSPs because of missing symbols in fsl_device_registers.h
+  if(NOT MCUX_IS_DSP)
+    zephyr_library_sources(mcux-sdk/devices/${MCUX_DEVICE_PATH}/drivers/fsl_power.c)
+  endif()
   zephyr_library_sources(mcux-sdk/devices/${MCUX_DEVICE_PATH}/drivers/fsl_reset.c)
 endif()
 
+# i.MXRT500 and i.MXRT600 have DSPs - DSP support expected.
+# Don't include DSP support while building for DSP cores, as it 
+# doesn't serve any purpose and depends on fsl_power.c.
+if (${MCUX_DEVICE} MATCHES "MIMXRT6|MIMXRT5" AND NOT MCUX_IS_DSP)
+    zephyr_library_sources(mcux-sdk/devices/${MCUX_DEVICE_PATH}/drivers/fsl_dsp.c)
+endif()
```

Comment:
Why do you need this for the non-DSP core?

Reply:
@mmahadevan108 Because it contains routines enabling controlling the DSP core from the ARM side. See, for example, the `xaf_usb_demo` example in the MCUXpresso SDK for whatever platform is it enabled. This PR is related to: https://github.com/zephyrproject-rtos/zephyr/pull/65229